src/main/java/com/gooddata/executeafm/afm/OverPeriodMeasureDefinition.java

```diff
@@ -0,0 +1,161 @@
+        this.dateAttributes = notEmpty(dateAttributes, "'dateAttributes' list must not be empty!");
+    }
+
+    /**
+     * The method is not supported by the object.
+     * Use {@link #getObjQualifiers()} instead.
+     *
+     * @return throws {@link UnsupportedOperationException}
+     *
+     * @throws UnsupportedOperationException
+     *         The exception is thrown every time the method is called.
+     * @deprecated Use {@link #getObjQualifiers()} instead.
+     */
+    @Override
+    @Deprecated
+    public ObjQualifier getObjQualifier() {
+        throw new UnsupportedOperationException("The method is not supported by the object!");
+    }
+
+    @Override
+    public Collection<ObjQualifier> getObjQualifiers() {
+        return this.dateAttributes.stream()
+                .map(attribute -> attribute.getAttribute())
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * The method is not supported by the object.
+     * Use {@link #withObjUriQualifiers(ObjQualifierConverter)} instead.
+     *
+     * @param uriQualifier
+     *         The URI object qualifier.
+     *
+     * @return throws {@link UnsupportedOperationException}
+     *
+     * @throws UnsupportedOperationException
+     *         The exception is thrown every time the method is called.
+     * @deprecated Use {@link #withObjUriQualifiers(ObjQualifierConverter)} instead.
+     */
+    @Override
+    @Deprecated
+    public MeasureDefinition withObjUriQualifier(final UriObjQualifier uriQualifier) {
+        throw new UnsupportedOperationException("The method is not supported by the object!");
+    }
+
+    @Override
+    public MeasureDefinition withObjUriQualifiers(final ObjQualifierConverter objQualifierConverter) {
+        notNull(objQualifierConverter, "'objQualifierConverter' must not be null!");
+        return new OverPeriodMeasureDefinition(measureIdentifier, copyAttributesWithUriQualifiers(objQualifierConverter));
+    }
+
+    private List<OverPeriodDateAttribute> copyAttributesWithUriQualifiers(final ObjQualifierConverter objQualifierConverter) {
+        return dateAttributes.stream()
+                .map(attribute -> copyWithUriQualifier(attribute, objQualifierConverter))
+                .collect(Collectors.toList());
+    }
+
+    private OverPeriodDateAttribute copyWithUriQualifier(final OverPeriodDateAttribute attribute, final ObjQualifierConverter objQualifierConverter) {
+        if (attribute.getAttribute() instanceof IdentifierObjQualifier) {
+            final IdentifierObjQualifier attributeIdentifier = (IdentifierObjQualifier) attribute.getAttribute();
+            return objQualifierConverter.convertToUriQualifier(attributeIdentifier)
+                    .map(uriObjQualifier -> new OverPeriodDateAttribute(uriObjQualifier, attribute.getPeriodsAgo()))
+                    .orElseThrow(() -> new IllegalArgumentException(format("The converter cannot convert identifier '%s'!", attributeIdentifier)));
```

Comment:
I would recommend more explicit message, something like "Supplied converter provides no conversion for..."

Reply:
Fixed, moved to `ObjIdentifierUtilities`.